src/OpenTelemetry/Trace/Export/BatchingActivityProcessor.cs

```diff
@@ -0,0 +1,247 @@
+// <copyright file="BatchingActivityProcessor.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using OpenTelemetry.Internal;
+
+namespace OpenTelemetry.Trace.Export
+{
+    /// <summary>
+    /// Implements processor that batches activities before calling exporter.
+    /// </summary>
+    public class BatchingActivityProcessor : ActivityProcessor, IDisposable
+    {
+        private const int DefaultMaxQueueSize = 2048;
+        private const int DefaultMaxExportBatchSize = 512;
+        private static readonly TimeSpan DefaultScheduleDelay = TimeSpan.FromMilliseconds(5000);
+        private readonly ConcurrentQueue<Activity> exportQueue;
+        private readonly int maxQueueSize;
+        private readonly int maxExportBatchSize;
+        private readonly TimeSpan scheduleDelay;
+        private readonly ActivityExporter exporter;
+        private readonly List<Activity> batch = new List<Activity>();
+        private CancellationTokenSource cts;
+        private volatile int currentQueueSize;
+        private bool stopping = false;
```

Comment:
Where is [`exporterTimeoutMillis`](https://github.com/open-telemetry/opentelemetry-specification/blob/6f47d037b52a9892570a0b0964d6b12f5891a0a3/specification/trace/sdk.md#batching-processor)?

Reply:
Added exporterTimeoutMillis

Reply:
Since the type is `TimeSpan` instead of int/long, I guess it is fine to call it `exporterTimeout`? Or if we want to call it `exporterTimeoutMillis`, probably need to change the type to int/long.

Reply:
`TimeSpan` is easier for consumers to set it, will modify the code to use `exporterTimeout`.